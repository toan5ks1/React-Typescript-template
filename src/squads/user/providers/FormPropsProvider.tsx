import { createContext, PropsWithChildren, useContext } from "react";

export interface FormPropsContextValue {
    readOnly: boolean;
}

const FormPropsContext = createContext<FormPropsContextValue>({ readOnly: false });

const { Provider } = FormPropsContext;

export const FormPropsProvider = ({
    children,
    readOnly,
}: PropsWithChildren<{ readOnly: boolean }>) => {
    return <Provider value={{ readOnly }}>{children}</Provider>;
};

export default FormPropsContext;
export const useFormProps = () => useContext(FormPropsContext);
