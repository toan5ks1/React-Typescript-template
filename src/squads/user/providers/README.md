### We abstract the provider into an isolated component, can be hard to test but parent component don't need to care about what the provider does
