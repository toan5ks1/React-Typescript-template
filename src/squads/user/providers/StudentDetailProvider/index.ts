export * from "./StudentDetailProvider";
export { default } from "./StudentDetailProvider";
