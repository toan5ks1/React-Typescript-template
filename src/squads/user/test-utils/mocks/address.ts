import { StudentHomeAddressFormProps, UserAddress } from "src/squads/user/common/types";

export const mockStudentHomeAddress: StudentHomeAddressFormProps = {
    postalCode: "70000",
    prefecture: "prefecture_id",
    city: "千代田区",
    firstStreet: "千代田",
    secondStreet: "１−１",
};

export const mockUserAddressList: UserAddress[] = [...Array(5).keys()].map((value) => {
    return {
        user_id: `user_id_${value}`,
        user_address_id: `user_address_id_${value}`,
        postal_code: `postal_code_${value}`,
        city: `city_${value}`,
        first_street: `first_street_${value}`,
        second_street: `second_street_${value}`,
        prefecture: {
            prefecture_id: `prefecture_id_${value}`,
            name: `prefecture_name_${value}`,
        },
    };
});
