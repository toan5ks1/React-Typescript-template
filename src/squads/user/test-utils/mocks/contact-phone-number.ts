import { StudentContactPhoneNumberFormProps } from "src/squads/user/common/types";

export const mockStudentContact: StudentContactPhoneNumberFormProps = {
    studentPhoneNumber: "0909673374",
    homePhoneNumber: "0909673374",
    parentPrimaryPhoneNumber: "0909673374",
    parentSecondaryPhoneNumber: "0909673374",
    preferredContactNumber: "Student Phone Number",
};
