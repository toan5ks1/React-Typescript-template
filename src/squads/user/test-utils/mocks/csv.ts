import { ArrayElement } from "src/common/constants/types";
import { NsUsermgmtStudentImportService } from "src/squads/user/service/usermgmt/student-import-service-usermgmt/types";

import {
    GenerateImportStudentTemplateResponse,
    ImportStudentResponse,
} from "manabuf/usermgmt/v2/student_pb";

export const importStudentTemplate =
    "Ym1GdFpTeGxiV0ZwYkN4bGJuSnZiR3h0Wlc1MFgzTjBZWFIxY3l4bmNtRmtaU3h3YUc5dVpWOXVkVzFpWlhJc1ltbHlkR2hrWVhrc1oyVnVaR1Z5TEd4dlkyRjBhVzl1Q25OMGRXUmxiblFnYm1GdFpTeHpkSFZrWlc1MFFHVnRZV2xzTG1OdmJTd3hMREVzTVRJek5EVTJOemc1TERJd01EQXZNREV2TURFc01TeE1NREU3VERBeQ==";

export const mockGenerateImportStudentTemplateResp = () => {
    const resp = new GenerateImportStudentTemplateResponse();
    resp.setData(importStudentTemplate);
    return resp;
};

export const mockImportStudentResponse = (
    error?: ArrayElement<NsUsermgmtStudentImportService.ImportStudentResp["errorsList"]>
) => {
    const resp = new ImportStudentResponse();
    const respError = new ImportStudentResponse.ImportStudentError();
    if (error) {
        respError.setError(error.error);
        respError.setRowNumber(error.rowNumber);
        resp.setErrorsList([respError]);
    }
    return resp;
};
