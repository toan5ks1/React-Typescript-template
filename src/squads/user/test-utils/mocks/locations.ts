import {
    flatTreeLocationByRecursive,
    getNodeTreeByRecursive,
} from "src/squads/user/common/helpers/tree-locations";
import { LocationInformation, LocationInformationHasura } from "src/squads/user/common/types";

import type {
    NodeLocationProps,
    TreeLocationProps,
} from "src/squads/user/hooks/useMapTreeLocations";

export const mockTreeLocations: TreeLocationProps[] = [
    {
        locationId: "org_1",
        name: "Organization",
        locationType: "0",
        parentLocationId: "",
        accessPath: "org_1",
        level: 0,
        children: [
            {
                locationId: "brand_1",
                name: "Individual Brand",
                locationType: "1",
                parentLocationId: "org_1",
                accessPath: "org_1/brand_1",
                level: 1,
                children: [
                    {
                        locationId: "prefecture_1_1",
                        name: "Nara prefecture",
                        locationType: "2",
                        parentLocationId: "brand_1",
                        accessPath: "org_1/brand_1/prefecture_1_1",
                        level: 2,
                        children: [
                            {
                                locationId: "center_1",
                                name: "Location 1",
                                locationType: "3",
                                parentLocationId: "prefecture_1_1",
                                accessPath: "org_1/brand_1/prefecture_1_1/center_1",
                                level: 3,
                                children: [],
                            },
                            {
                                locationId: "center_2",
                                name: "Location 2",
                                locationType: "3",
                                parentLocationId: "prefecture_1_1",
                                accessPath: "org_1/brand_1/prefecture_1_1/center_2",
                                level: 3,
                                children: [],
                            },
                            {
                                locationId: "center_3",
                                name: "Location 3",
                                locationType: "3",
                                parentLocationId: "prefecture_1_1",
                                accessPath: "org_1/brand_1/prefecture_1_1/center_3",
                                level: 3,
                                children: [],
                            },
                            {
                                locationId: "center_4",
                                name: "Location 4",
                                locationType: "3",
                                parentLocationId: "prefecture_1_1",
                                accessPath: "org_1/brand_1/prefecture_1_1/center_4",
                                level: 3,
                                children: [],
                            },
                            {
                                locationId: "center_5",
                                name: "Location 5",
                                locationType: "3",
                                parentLocationId: "prefecture_1_1",
                                accessPath: "org_1/brand_1/prefecture_1_1/center_5",
                                level: 3,
                                children: [],
                            },
                            {
                                locationId: "center_6",
                                name: "Location 6",
                                locationType: "3",
                                parentLocationId: "prefecture_1_1",
                                accessPath: "org_1/brand_1/prefecture_1_1/center_6",
                                level: 3,
                                children: [],
                            },
                            {
                                locationId: "center_7",
                                name: "Location 7",
                                locationType: "3",
                                parentLocationId: "prefecture_1_1",
                                accessPath: "org_1/brand_1/prefecture_1_1/center_7",
                                level: 3,
                                children: [],
                            },
                            {
                                locationId: "center_8",
                                name: "Location 8",
                                locationType: "3",
                                parentLocationId: "prefecture_1_1",
                                accessPath: "org_1/brand_1/prefecture_1_1/center_8",
                                level: 3,
                                children: [],
                            },
                            {
                                locationId: "center_9",
                                name: "Location 9",
                                locationType: "3",
                                parentLocationId: "prefecture_1_1",
                                accessPath: "org_1/brand_1/prefecture_1_1/center_9",
                                level: 3,
                                children: [],
                            },
                            {
                                locationId: "center_10",
                                name: "Location 10",
                                locationType: "3",
                                parentLocationId: "prefecture_1_1",
                                accessPath: "org_1/brand_1/prefecture_1_1/center_10",
                                level: 3,
                                children: [],
                            },
                            {
                                locationId: "center_11",
                                name: "Location 11",
                                locationType: "3",
                                parentLocationId: "prefecture_1_1",
                                accessPath: "org_1/brand_1/prefecture_1_1/center_11",
                                level: 3,
                                children: [],
                            },
                            {
                                locationId: "center_12",
                                name: "Location 12",
                                locationType: "3",
                                parentLocationId: "prefecture_1_1",
                                accessPath: "org_1/brand_1/prefecture_1_1/center_12",
                                level: 3,
                                children: [],
                            },
                            {
                                locationId: "center_13",
                                name: "Location 13",
                                locationType: "3",
                                parentLocationId: "prefecture_1_1",
                                accessPath: "org_1/brand_1/prefecture_1_1/center_13",
                                level: 3,
                                children: [],
                            },
                        ],
                    },
                    {
                        locationId: "prefecture_1_2",
                        name: "Osaka prefecture",
                        locationType: "2",
                        parentLocationId: "brand_1",
                        accessPath: "org_1/brand_1/prefecture_1_2",
                        level: 2,
                        children: [
                            {
                                locationId: "center_14",
                                name: "Location 14",
                                locationType: "3",
                                parentLocationId: "prefecture_1_2",
                                accessPath: "org_1/brand_1/prefecture_1_2/center_14",
                                level: 3,
                                children: [],
                            },
                            {
                                locationId: "center_15",
                                name: "Location 15",
                                locationType: "3",
                                parentLocationId: "prefecture_1_2",
                                accessPath: "org_1/brand_1/prefecture_1_2/center_15",
                                level: 3,
                                children: [],
                            },
                        ],
                    },
                ],
            },
            {
                locationId: "brand_2",
                name: "Group Brand",
                locationType: "1",
                parentLocationId: "org_1",
                accessPath: "org_1/brand_2",
                level: 1,
                children: [
                    {
                        locationId: "prefecture_2_1",
                        name: "Test prefecture",
                        locationType: "prefecture",
                        parentLocationId: "brand_2",
                        accessPath: "org_1/brand_2/prefecture_2_1",
                        level: 2,
                        children: [
                            {
                                locationId: "center_16",
                                name: "Location 16",
                                locationType: "3",
                                parentLocationId: "prefecture_2_1",
                                accessPath: "org_1/brand_2/prefecture_2_1/center_16",
                                level: 3,
                                children: [],
                            },
                            {
                                locationId: "center_17",
                                name: "Location 17",
                                locationType: "3",
                                parentLocationId: "prefecture_2_1",
                                accessPath: "org_1/brand_2/prefecture_2_1/center_17",
                                level: 3,
                                children: [],
                            },
                        ],
                    },
                ],
            },
        ],
    },
];

export const mockLowestNodeLocation: TreeLocationProps = getNodeTreeByRecursive(
    mockTreeLocations[0],
    "center_13"
);

export const mockListLocationsChecked: LocationInformation[] = [mockLowestNodeLocation];

export const mockBranchLocation: TreeLocationProps = getNodeTreeByRecursive(
    mockTreeLocations[0],
    "prefecture_1_1"
);

export const mockFlatLocations: NodeLocationProps[] = flatTreeLocationByRecursive(
    mockTreeLocations[0]
);

export const getMockLocations = (length: number = 20, start: number = 0): LocationInformation[] => {
    return Array(length)
        .fill("")
        .map((_, index) => ({
            locationId: `center_${index + 1 + start}`,
            name: `Location ${index + 1 + start}`,
            parentLocationId: `prefecture_1_1`,
            locationType: "center",
            accessPath: "",
        }));
};

export const getMockLocationsHasura = (
    length: number = 20,
    start: number = 0
): LocationInformationHasura[] => {
    return Array(length)
        .fill("")
        .map((_, index) => ({
            location_id: `center_${index + start}`,
            name: `Location ${index + 1 + start}`,
            parent_location_id: `prefecture_1_1`,
            location_type: "center",
            access_path: "",
            is_archived: false,
        }));
};
