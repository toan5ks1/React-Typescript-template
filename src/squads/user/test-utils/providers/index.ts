export { default as TestThemeProvider } from "./TestThemeProvider";
export { default as TestCommonAppProvider } from "./TestCommonAppProvider";
export * from "./TestQueryWrapper";
export { default as TestHookForm } from "./TestHookForm";
export { default as TestHookFormProvider } from "./TestHookFormProvider";
export { default as TestingRouter } from "./TestingRouter";
export { default as AppProvider } from "./AppProvider";
export { default as TestStudentDetailProvider } from "./TestStudentDetailProvider";
