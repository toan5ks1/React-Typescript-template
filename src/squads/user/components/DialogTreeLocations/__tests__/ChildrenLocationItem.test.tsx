import { mockLowestNodeLocation } from "src/squads/user/test-utils/mocks/locations";
import { TestCommonAppProvider } from "src/squads/user/test-utils/providers";

import ChildrenLocationItem from "../ChildrenLocationItem";

import { render, screen, within } from "@testing-library/react";

describe("<ChildrenLocationItem/>", () => {
    const renderComponent = (checked: boolean = true) => {
        return render(
            <TestCommonAppProvider>
                <ChildrenLocationItem
                    option={mockLowestNodeLocation}
                    keyShowValue="name"
                    checked={checked}
                />
            </TestCommonAppProvider>
        );
    };

    it("should match snapshot on checked", () => {
        const { container } = renderComponent();
        expect(container).toMatchSnapshot();
    });
    it("should be checked checkbox", () => {
        renderComponent();

        const containerCheckbox = screen.getByTestId("ItemLocation__checkbox");

        const inputCheckbox = within(containerCheckbox).getByRole("checkbox");

        expect(inputCheckbox).toBeChecked();
    });
    it("should not be checked checkbox", () => {
        renderComponent(false);

        const containerCheckbox = screen.getByTestId("ItemLocation__checkbox");

        const inputCheckbox = within(containerCheckbox).getByRole("checkbox");

        expect(inputCheckbox).not.toBeChecked();
    });
});
