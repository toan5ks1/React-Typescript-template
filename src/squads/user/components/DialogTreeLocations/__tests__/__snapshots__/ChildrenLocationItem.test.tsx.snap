// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChildrenLocationItem/> should match snapshot on checked 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  padding: 9px;
  border-radius: 50%;
  color: #757575;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  background-color: rgba(33, 150, 243, 0.04);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-checked,
.emotion-0.MuiCheckbox-indeterminate {
  color: #2196F3;
}

.emotion-0.Mui-disabled {
  color: #BDBDBD;
}

.emotion-0.MuiCheckbox-indeterminate {
  color: #757575;
}

.emotion-0.CheckboxBase-selector {
  padding: 9px;
}

.emotion-1 {
  cursor: inherit;
  position: absolute;
  opacity: 0;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-3 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-4 {
  margin: 0;
  font-size: 0.875rem;
  font-weight: 400;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.43;
}

<div>
  <span
    class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked CheckboxBase-selector emotion-0"
    data-testid="ItemLocation__checkbox"
  >
    <input
      checked=""
      class="PrivateSwitchBase-input emotion-1"
      data-indeterminate="false"
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
      data-testid="CheckBoxIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root emotion-3"
    />
  </span>
  <p
    class="MuiTypography-root MuiTypography-body2 emotion-4"
    data-testid="ItemLocation__label"
  >
    Location 13
  </p>
</div>
`;
